Update three.js to r118

fredludlow opened the pull request:
Hopefully this finally fixes #728 (by leapfrogging three.js to version r118).

Many thanks to @garyo for the pointers on getting this working on IE11 (WebGl1).

I have tested this on Windows 10 (Chrome, FF, IE11, Edge), Ubuntu (Chrome, FF) Android (chrome, FF) and iOS (safari mobile) . It would be really helpful if someone could check if this renders on Safari (on a Mac desktop) as I don't have access to that.

https://fredludlow.com/ngl/three118/examples/webapp.html?script=representation/tube-bfactor

ppillot commented:
I've tested it with Safari on MacOS, and there is an issue with the rendering of the sticks
<img width="609" alt="Capture d’écran, le 2020-07-08 à 08 14 08" src="https://user-images.githubusercontent.com/3930632/86917396-07e11980-c0f3-11ea-8ddb-2fb96d63fe2b.png">

garyo commented:
I've tested this as well, and at least in Chrome on Windows it's working great (once I updated my own code to three r118 and rebuilt everything). That said, I do have some of my own commits on top of this one so it's possible I've fixed something, but I don't see anything relevant. @ppillot, perhaps it's a WebGL1 vs. WebGL2 context thing -- in r118, three.js switched the default:

> "WebGLRenderer automatically creates a WebGL 2 rendering context now (and fallbacks to WebGL 1 if necessary). If your project can only use WebGL 1, you can use WebGL1Renderer."

fredludlow commented:
@garyo, This might be tricky to debug, so if you want to rebase your other
prs on current master rather than this then we can get those merged quicker

On Fri, 10 Jul 2020, 18:22 GaryO, <notifications@github.com> wrote:

> I've tested this as well, and at least in Chrome on Windows it's working
> great (once I updated my own code to three r118 and rebuilt everything).
> That said, I do have some of my own commits on top of this one so it's
> possible I've fixed something, but I don't see anything relevant. @ppillot
> <https://github.com/ppillot>, perhaps it's a WebGL1 vs. WebGL2 context
> thing -- in r118, three.js switched the default:
>
> "WebGLRenderer automatically creates a WebGL 2 rendering context now (and
> fallbacks to WebGL 1 if necessary). If your project can only use WebGL 1,
> you can use WebGL1Renderer."
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/arose/ngl/pull/764#issuecomment-656790669>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAG33N4Q37FJKOWVB5QEEYLR25E3ZANCNFSM4OUP3THQ>
> .
>

garyo commented:
Agreed -- working on it today. Need to clean up some internal mess first ;-)

garyo commented:
My additional PRs are now posted: #738, #739 and #748 are force-pushes based on this PR (actually fredludlow/jussty-728, 47a4a021), and #765, #766 and #767 are new ones. They are all stacked on one another:
```
% git ogl
┯ ffdb06f3 - Give nice names to mesh and wireframe graphics.. Gary O.. 21 min..  ✓ fixes 💻    minor/name-groups 📡 💻
┿ 13949a95 - Make trackball controls work with camera not j..          22 min..  feature/generalize-camera-position 📡
┿ 136b410e - Generalize camera to not have to be along -z a..          28 min..
┿ f56a7756 - Reset rendering state if error found while ren..          32 min..  fix/reset-render-state-on-error 📡 💻
┿ 139b461f - More clipping handling; see PR #656                       34 min..  fix/clipping-in-camera-clip-mode 📡 💻
┿ b9ecf3d9 - Export PickingProxy type                                  36 min..  fix/export-pickingproxy 📡 💻
┿ fb16f822 - Update & clean up dependencies, fix rollup con..          40 min..  fix/cleanup-package-deps 📡 💻
┿ 47a4a021 - Ensure the OES_standard_derivatives extension .. Fred L.. 2 days..  fredludlow/jussty-728 💻
```

@ppillot you're welcome to try out my branch (just use `minor/name-groups`, it's the most recent). I suspect it won't work with r118 but three r117 will. Let's see.

fredludlow commented:
@ppillot  - Thanks for testing that, I'm not totally sure I've understood why this is happening but have some ideas and will try to fix. Before I go too mad looking at it, could you just confirm whether the same example is working in current master?
https://fredludlow.com/ngl/master/examples/webapp.html?script=representation/tube-bfactor

fredludlow commented:
Aha, I can reproduce that bug by forcing webgl1 on linux.. (using three.WebGL1Renderer)

We also need to not check for certain extensions in WebGL2 (since many are built in so testing for them gives a false negative)

ppillot commented:
> Before I go too mad looking at it, could you just confirm whether the same example is working in current master?
> https://fredludlow.com/ngl/master/examples/webapp.html?script=representation/tube-bfactor

I can confirm the aforementioned link works fine for me in Safari

fredludlow commented:
Thanks again @ppillot - could you possibly test another one for me (and indeed anyone else who can report). I have definitely fixed some bugs in this version, fingers crossed its sufficient to get it working in Safari:

https://fredludlow.com/ngl/three118/examples/webapp.html?script=representation/tube-bfactor

garyo commented:
Your latest works in Safari for me -- nice!
Safari 13.1.1, MacOS Catalina 10.15.5.
Tried a bunch of the examples, they all look good too.

[image: image.png]

On Mon, Jul 13, 2020 at 11:15 AM Fred Ludlow <notifications@github.com>
wrote:

> Thanks again @ppillot <https://github.com/ppillot> - could you possibly
> test another one for me (and indeed anyone else who can report). I have
> definitely fixed some bugs in this version, fingers crossed its sufficient
> to get it working in Safari:
>
>
> https://fredludlow.com/ngl/three118/examples/webapp.html?script=representation/tube-bfactor
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/arose/ngl/pull/764#issuecomment-657620301>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABCFR6KQM3ZJWJFILCSJVDR3MQIJANCNFSM4OUP3THQ>
> .
>


-- 
Gary Oberbrunner
Founder & CEO
Dark Star Systems, Inc <https://darkstarsystems.com>.

fredludlow commented:
I think this is good to merge now, but it needs a review. @garyo would you mind taking a look?

I haven't included the built files, but if you think I should let me know and I'll push them

garyo commented:
As for the built files, I've never been a fan of checking those in -- makes the repo unnecessarily huge. I don't know how others use it though, so probably best to, at least for a big change like this.

fredludlow commented:
Looking at the commit history for /dist, Alex used to commit those files on a dev.xx version change (and these get pushed to npm and are available (I think) as npm install ngl@next). Let's figure that bit out separately.
